### PR TITLE
Use Scala 2.11 by default

### PR DESCRIPTION
**Pull request by vinooganesh**
This PR reverts back to using Scala 2.11

* Revert "Fix distribution publish to scala 2.12 #478"
* Revert "[SPARK-25956] Make Scala 2.12 as default Scala version in Spark 3.0"

### CODE REVIEW

File: dev/make-distribution.sh [hunk 1]
```diff
@@ -153,7 +153,7 @@ fi
 echo "Spark version is $VERSION"
 VERSION_SET=$("$MVN" versions:set -DnewVersion=$VERSION | tail -n 1)
 if [ "$MAKE_TGZ" == "true" ]; then
-  echo "Making spark-dist-$NAME-$VERSION.tgz"
```
Comment (robert3005) [inline dev/make-distribution.sh]: I think this was right and previous thing was wrong

File: dists/hadoop-palantir-bom/pom.xml [hunk 2]
```diff
@@ -107,7 +107,7 @@
             </dependency>
             <dependency>
                 <groupId>org.apache.spark</groupId>
-                <artifactId>spark-avro_${scala.binary.version}</artifactId>
```
Comment (robert3005) [inline dists/hadoop-palantir-bom/pom.xml]: why?

File: dists/hadoop-palantir/pom.xml [hunk 3]
```diff
@@ -57,7 +57,7 @@
     </dependency>
     <dependency>
       <groupId>org.apache.spark</groupId>
-      <artifactId>spark-avro_${scala.binary.version}</artifactId>
```
Comment (robert3005) [inline dists/hadoop-palantir/pom.xml]: why?

File: docs/building-spark.md [hunk 4]
```diff
@@ -96,9 +96,9 @@ It's possible to build Spark submodules using the `mvn -pl` option.
 
 For instance, you can build the Spark Streaming module using:
 
-    ./build/mvn -pl :spark-streaming_{{site.SCALA_BINARY_VERSION}} clean install
```
Comment (robert3005) [inline docs/building-spark.md]: changes to this file are probably wrong

File: dev/make-distribution.sh (hunk 1 above)
Comment (vinooganesh) [reply dev/make-distribution.sh]: wait you're saying `echo "Making spark-dist-$NAME-$VERSION.tgz"` is correct, right?

File: dists/hadoop-palantir-bom/pom.xml (hunk 2 above)
Comment (vinooganesh) [reply dists/hadoop-palantir-bom/pom.xml]: seems like an artifact of the revert

File: dev/deps/spark-deps-hadoop-palantir [hunk 5]
```diff
@@ -173,28 +171,28 @@ orc-shims-1.5.3.jar
 oro-2.0.8.jar
 osgi-resource-locator-1.0.1.jar
 paranamer-2.8.jar
-parquet-column-1.12.0-palantir.4.jar
-parquet-common-1.12.0-palantir.4.jar
-parquet-encoding-1.12.0-palantir.4.jar
-parquet-format-structures-1.12.0-palantir.4.jar
-parquet-hadoop-1.12.0-palantir.4.jar
-parquet-jackson-1.12.0-palantir.4.jar
+parquet-column-1.12.0-palantir.1.jar
```
Comment (vinooganesh) [inline dev/deps/spark-deps-hadoop-palantir]: need to revert these

**Pull request merged**